app/src/main/java/org/jellyfin/androidtv/details/MovieDetailsFragment.kt

```diff
@@ -50,15 +50,17 @@ class MovieDetailsFragment(item: Movie) : BaseDetailsFragment<Movie>(item) {
 
 			if (item.canResume) add(ResumeAction(context!!, item))
 			add(PlayFromBeginningAction(context!!, item))
-			// TODO: Bring this back once we have a more understandable queue implementation for users
-			//add(ToggleWatchedAction(context!!, item))
+			add(ToggleWatchedAction(context!!, item))
 			add(ToggleFavoriteAction(context!!, item))
-			add(AddToQueueAction(context!!, item))
+			// TODO: Bring this back once we have a more understandable queue implementation for users
+			//add(AddToQueueAction(context!!, item))
 
 			// Menu with more actions
-			add(SecondariesPopupAction(context!!).apply {
-				add(DeleteAction(context!!, item) { activity?.finish() })
-			})
+			if (TvApp.getApplication().currentUser.policy.enableContentDeletion) {
```

Comment:
This if statement should be for the DeleteAction only. I imagine you did it this way to the popup button wouldn't show when there's no items but we need to fix that in another way.

1. Extend action with a `isvisible` property or something similar
2. implement it for secondariespopupaction and check if the amount if children >= 1
2. When showing the actions, look if it's visible
4. that's it